runtime/src/accounts_db.rs

```diff
@@ -1397,8 +1426,52 @@ impl AccountsDB {
         hashes
     }
 
+    pub fn freeze_accounts(&self, ancestors: &HashMap<Slot, usize>, account_pubkeys: &[Pubkey]) {
+        let mut frozen_accounts = self.frozen_accounts.lock().unwrap();
+        for account_pubkey in account_pubkeys {
+            let (account, _slot) = self
+                .load_slow(ancestors, &account_pubkey)
+                .unwrap_or_default();
+            let frozen_account_info = FrozenAccountInfo {
+                hash: Self::hash_frozen_account_data(&account),
+                lamports: account.lamports,
+            };
+            warn!(
+                "Account {} is now frozen at lamports={}, hash={}",
+                account_pubkey, frozen_account_info.lamports, frozen_account_info.hash
+            );
+            frozen_accounts.insert(*account_pubkey, frozen_account_info);
+        }
+    }
+
     /// Store the account update.
     pub fn store(&self, slot_id: Slot, accounts: &[(&Pubkey, &Account)]) {
+        let mut frozen_accounts = self.frozen_accounts.lock().unwrap();
+        if !frozen_accounts.is_empty() {
+            for (account_pubkey, account) in accounts.iter() {
+                frozen_accounts
+                    .entry(**account_pubkey)
+                    .and_modify(|frozen_account_info| {
+                        if account.lamports < frozen_account_info.lamports {
+                            panic!(
+                                "Frozen account {} modified.  Lamports decreased from {} to {}",
+                                account_pubkey, frozen_account_info.lamports, account.lamports,
+                            )
+                        } else {
+                            frozen_account_info.lamports = account.lamports;
```

Comment:
Seems like it's not frozen, it's just strictly increasing.

Reply:
Yeah.  `frozen_data_with_strictly_increasing_lamports` is a mouthful, I've very open to better names.    Ideally it would be strictly frozen, but a system transfer cannot be blocked by the recipient so balance increases must be accepted

Reply:
yea. fair. monitored_account maybe, but that's somewhat vague. It's just a bit confusing at first exactly what frozen account means.

Reply:
donno, I think frozen is ok for now.   This really isn't an end-user feature.  It's just for our use until we're ok not having a super-minority on mainnet-beta, at which point we'll remove this feature entirely as it's no longer relevant.